versions-common/src/main/java/org/codehaus/mojo/versions/filtering/DependencyFilter.java

```diff
@@ -77,10 +79,51 @@ private boolean matchersMatch( Dependency dependency )
         return matchers.stream().anyMatch( m -> m.test( dependency ) );
     }
 
-    private TreeSet<Dependency> filterBy( Set<Dependency> dependencies, Predicate<Dependency> predicate )
+    private TreeSet<Dependency> filterBy( Collection<Dependency> dependencies, Predicate<Dependency> predicate )
     {
         return dependencies.stream()
                 .filter( predicate )
                 .collect( Collectors.toCollection( () -> new TreeSet<>( DependencyComparator.INSTANCE ) ) );
     }
+
+    /**
+     * Returns a set of dependencies filtered by the given include- and exclude filters.
+     * @param dependencies collection of dependencies to filter
+     * @param includes a list of dependency includes
+     * @param excludes a list of dependency excludes
+     * @param section if log is not null, dependency section name for the debug log
+     * @param log null or log to which debug information will be logged
+     * @return filtered set of dependencies
+     */
+    public static Set<Dependency> filterDependencies(
+            Collection<Dependency> dependencies,
+            List<String> includes,
+            List<String> excludes,
+            String section,
+            Log log
```

Comment:
We can use SLF4J directly is such cases

Reply:
I only reused the class created by Christoph Linder. The class is used in `DisplayDependencyUpdatesMojo`, not just in the enforcer filter. I think removing maven Log from here this would be out of scope of this PR.

Reply:
Or do you want to create a second invocation without the log, in which case an SLF4J instance would be used instead?

Reply:
ok, next time